### PR TITLE
view Donation API and its test Cases

### PR DESCRIPTION
**Pull request by raiyanfiraz**
1. Added View Donation History API
2. Added Test cases for View Donation History API

### CODE REVIEW

File: app/serializers.py
Comment (iamshreeni) [inline app/serializers.py:1]: changes need to be made

File: app/serializers.py [hunk 1]
```diff
@@ -356,7 +356,7 @@ def get_dropDate(self, obj):
         if obj.dropDate is not None:
             return obj.dropDate.strftime('%Y-%m-%d')
         else:
-            return '0000-00-00'
+            return 'drop Date not Available'
```
Comment (raiyanfiraz) [inline app/serializers.py:359]: Refactor the message

Comment (raiyanfiraz) [inline app/serializers.py:359]: Refactor message

**Review by sachinrohit — CHANGES_REQUESTED** [review]

File: app/views.py [hunk 2]
```diff
@@ -1126,5 +1126,51 @@ def post(self, request, format=None):
...
+
+    def get(self, request, format=None):
+        try:
+         
+            if request.user.id is not None:
+                userId= request.user.id
+                if Volunteer.objects.filter(id=userId).exists():
+                    user = Volunteer.objects.get(id=userId)
+                else:
+                    return Response({'success': False, 'message': 'user not found'})
```
[the remark this answers is not in the document]
Comment (raiyanfiraz) [reply app/views.py]: Have replaced it with request.user

**Pull request merged**